benchmarks/BaselinesTcp.cpp

```diff
@@ -0,0 +1,178 @@
+    }
+
+    accepting.store(true);
+
+    if ((sock = accept(
+        serverSock, reinterpret_cast<struct sockaddr*>(&addr), &addrlen)) <
+        0) {
+      perror("accept");
+      return;
+    }
+
+    accepted.store(true);
+
+    size_t sentBytes = 0;
+    while (sentBytes < loadSize) {
+      if (send(sock, message, msgLength, 0) !=
+          static_cast<ssize_t>(msgLength)) {
+        perror("send");
+        return;
+      }
+      sentBytes += msgLength;
+    }
+
+    close(sock);
+    close(serverSock);
+  });
+
+  while (!accepting) {
+    std::this_thread::yield();
+  }
+
+  int sock = socket(AF_INET, SOCK_STREAM, 0);
+  struct sockaddr_in addr;
+  socklen_t addrlen = sizeof(addr);
+  char message[MAX_MESSAGE_LENGTH];
+
+  std::memset(message, 0, sizeof(message));
+  std::memset(&addr, 0, sizeof(addr));
+
+  if (sock < 0) {
+    perror("connector socket");
+    return;
+  }
+
+  addr.sin_family = AF_INET;
+  addr.sin_addr.s_addr = inet_addr("127.0.0.1");
+  addr.sin_port = htons(PORT);
+  if (connect(sock, reinterpret_cast<struct sockaddr*>(&addr), addrlen) < 0) {
+    perror("connect");
+    return;
+  }
+
+  while (!accepted) {
+    std::this_thread::yield();
+  }
+
+  size_t receivedBytes = 0;
+  while (receivedBytes < loadSize) {
+    ssize_t recved = recv(sock, message, recvLength, 0);
+
+    if (recved < 0) {
+      perror("recv");
+      return;
+    }
+
+    receivedBytes += recved;
+  }
+
+  close(sock);
+  t.join();
+}
+
+void BM_Baseline_TCP_Throughput_100MB_s40B_r1024B(unsigned, unsigned) {
+  constexpr size_t loadSizeB = 100 * 1024 * 1024;
+  constexpr size_t sendSizeB = 40;
+  constexpr size_t receiveSizeB = 1024;
+  BM_Baseline_TCP_SendReceive(loadSizeB, sendSizeB, receiveSizeB);
+}
+void BM_Baseline_TCP_Throughput_100MB_s40B_r4096B(unsigned, unsigned) {
+  constexpr size_t loadSizeB = 100 * 1024 * 1024;
+  constexpr size_t sendSizeB = 40;
+  constexpr size_t receiveSizeB = 4096;
+  BM_Baseline_TCP_SendReceive(loadSizeB, sendSizeB, receiveSizeB);
+}
+void BM_Baseline_TCP_Throughput_100MB_s80B_r4096B(unsigned, unsigned) {
+  constexpr size_t loadSizeB = 100 * 1024 * 1024;
+  constexpr size_t sendSizeB = 80;
+  constexpr size_t receiveSizeB = 4096;
+  BM_Baseline_TCP_SendReceive(loadSizeB, sendSizeB, receiveSizeB);
+}
+void BM_Baseline_TCP_Throughput_100MB_s4096B_r4096B(unsigned, unsigned) {
+  constexpr size_t loadSizeB = 100 * 1024 * 1024;
+  constexpr size_t sendSizeB = 4096;
+  constexpr size_t receiveSizeB = 4096;
+  BM_Baseline_TCP_SendReceive(loadSizeB, sendSizeB, receiveSizeB);
+}
+
+BENCHMARK_PARAM(BM_Baseline_TCP_Throughput_100MB_s40B_r1024B, 1);
+BENCHMARK_PARAM(BM_Baseline_TCP_Throughput_100MB_s40B_r4096B, 1);
+BENCHMARK_PARAM(BM_Baseline_TCP_Throughput_100MB_s80B_r4096B, 1);
+BENCHMARK_PARAM(BM_Baseline_TCP_Throughput_100MB_s4096B_r4096B, 1);
+
+void BM_Baseline_TCP_Latency_1M_msgs_32B(unsigned, unsigned) {
+  constexpr size_t messageSizeB = 32;
+  constexpr size_t loadSizeB = 1000000 * messageSizeB;
```

Comment:
nit: we can put point in the integer values for easy read.
1000000 => 1.000.000